src/sentry/spans/grouping/strategy/base.py

```diff
@@ -135,6 +135,32 @@ def loose_normalized_db_span_in_condition_strategy(span: Span) -> Optional[Seque
     return [cleaned]
 
 
+def join_regexes(regexes: Sequence[str]) -> str:
+    return r"(?:" + r")|(?:".join(regexes) + r")"
+
+
+DB_PARAMETRIZATION_PATTERN = re.compile(
+    join_regexes(
+        [
+            r"'(?:[^']|'')*?(?:\\'.*|'(?!'))",  # single-quoted strings
+            r"-?\b(?:[0-9]+\.)?[0-9]+(?:[eE][+-]?[0-9]+)?\b",  # numbers
+            r"\b(?:true|false)\b",  # booleans
+        ]
+    )
```

Comment:
iirc it's been a bit, and I think our input is bounded enough and it looks like just capture groups so that it won't be an issue, but in general regex + lots of data can cause performance overhead, especially since for a single event we can run up into the hundreds or thousands of sql statements to compare this against. Even though this is in 'dry-run' mode the performance impact still applies to production, I'd maybe just check quickly locally if you have one of those long non-truncated sql statements.

Reply:
Sure, I'll find some SQL-heavy events to benchmark against locally :+1: Good call, thanks!

Reply:
yep, was just a concern since it's one of those possible unintentional leaks even though this is supposed to be 'dry-run' so brought it having hit regex being troublesome in the past.